src/os/bluestore/NVMEDevice.cc

```diff
@@ -986,10 +986,40 @@ void NVMEDevice::aio_submit(IOContext *ioc)
     ioc->num_pending -= pending;
     assert(ioc->num_pending.load() == 0);  // we should be only thread doing this
     // Only need to push the first entry
-  if(queue_id == -1)
-    queue_id = ceph_gettid();
-    driver->get_queue(queue_id)->queue_task(t, pending);
+    if(queue_id == -1)
+      queue_id = ceph_gettid();
     ioc->nvme_task_first = ioc->nvme_task_last = nullptr;
+    driver->get_queue(queue_id)->queue_task(t, pending);
+  }
+}
+
+static void write_split(
+    NVMEDevice *dev,
+    uint64_t off,
+    bufferlist &bl,
+    IOContext *ioc)
+{
+  uint64_t remain_len = bl.length(), begin = 0, write_size;
+  Task *t, *first, *last;
+  uint64_t split_size = 131072; //128KB
+
+  while (remain_len > 0) {
+    write_size = std::min(remain_len, (uint64_t)split_size);
+    t = new Task(dev, IOCommand::WRITE_COMMAND, off + begin, write_size);
```

Comment:
maybe in future, let's use a Task pool instead of new each time. New and delete are not effective.

Reply:
That should be another patch